internal/codegen/queries.go

```diff
@@ -247,10 +251,37 @@ func BuildQueriesFile(engine string, config core.Config, queryFilename string, q
 			methodBody.WriteIndentedString(2, "var stmt = conn.prepareStatement("+q.MethodName+");\n")
 		}
 
+		queryInput := ""
+
+		// input method for query params limit
+		if len(q.Args) > config.QueryParameterLimit && config.QueryParameterLimit != 0 {
+			queryInput = queryInputName(q)
+
+			body.WriteString("\n")
+			body.WriteIndentedString(1, "public record "+queryInput+"(\n")
+			for i, arg := range q.Args {
+				imps, err := body.writeParameter(arg.JavaType, arg.Name, nonNullAnnotation, nullableAnnotation)
+				if err != nil {
+					return "", nil, err
+				}
+				if imps != nil {
+					imports = append(imports, imps...)
+				}
+
+				if i != len(q.Args)-1 {
+					body.WriteString(",\n")
+				}
+			}
+			body.WriteString("\n")
+			body.WriteIndentedString(1, ") {}\n")
+		}
+
 		// write the method signature
 		body.WriteString("\n")
 		body.WriteIndentedString(1, fmt.Sprintf("public %s %s(", returnType, q.MethodName))
-		if len(q.Args) > 0 {
+		if len(q.Args) > config.QueryParameterLimit && config.QueryParameterLimit != 0 {
+			body.WriteString(fmt.Sprintf("%s %sinput) throws SQLException{\n", queryInput, queryInput))
```

Comment:
```suggestion
			body.WriteString(fmt.Sprintf("%s input) throws SQLException{\n", queryInput))
```

Lets make the argument name just `input`